api/base_controller.go

```diff
@@ -509,22 +516,76 @@ func cleanObject(object types.Object) {
 		secured.Sanitize()
 	}
 }
+func getResourceIds(resources types.ObjectList) []string {
+	var resourceIds []string
+	for i := 0; i < resources.Len(); i++ {
+		resource := resources.ItemAt(i)
+		resourceIds = append(resourceIds, resource.GetID())
+	}
+	return resourceIds
+}
+
+func attachLastOperations(ctx context.Context, resources types.ObjectList, repository storage.Repository) error {
+	lastOperationsMap, err := getLastOperations(ctx, getResourceIds(resources), repository)
+
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < resources.Len(); i++ {
+		resource := resources.ItemAt(i)
+		if LastOp, ok := lastOperationsMap[resource.GetID()]; ok {
+			LastOp.TransitiveResources = nil
+			resource.SetLastOperation(LastOp)
+		}
+	}
+
+	return nil
+}
+
+func getLastOperations(ctx context.Context, resourceIDs []string, repository storage.Repository) (map[string]*types.Operation, error) {
+
```

Comment:
remove?

Reply:
getLastOperations is used in attachLastOperations &  attachLastOperation, that's the common code that use's the named query for both methods.

Reply:
sorry, meant the empty line :)